### PR TITLE
Different tile size for different tile layer

### PR DESCRIPTION
**Pull request by YosuaHamonangan**
I need this because I use different tile size for different tile layer in Melon Editor

### CODE REVIEW

File: src/level/TMXLayer.js [hunk 1]
```diff
@@ -65,8 +65,8 @@
             this._super(me.Renderable, "init", [0, 0, 0, 0]);
 
             // tile width & height
-            this.tilewidth  = tilewidth;
-            this.tileheight = tileheight;
+            this.tilewidth  = tilewidth || data.tilewidth;
+            this.tileheight = tileheight || data.tilewidth;
```
Comment (obiot) [inline src/level/TMXLayer.js]: is there not a typo though here, it should be `tileheight` ?

**Comment by obiot** [comment]
I'm not sure I understand though, if `tilewidth` and `tileheight` are anyway parameters of the constructor, why add new ones in `data` ?

File: src/level/TMXLayer.js (hunk 1 above)
Comment (YosuaHamonangan) [reply src/level/TMXLayer.js]: oh yeah its a typo

**Comment by YosuaHamonangan** [comment]
because the data variable contains the information for individual tile layer, so if I want to have different tile size for each layer, the best place to put the size information is in that data variable. 

and in my file format, I leave the original tilewidth and tileheight undefined

**Comment by obiot** [comment]
hi, will you update the change request ? as well I think It would make more sense to prioritise `data` over the given parameters.

I'm I am correct, Tiled used to have that features where it was possible to have layers with different tile size, and if one day it comes back, then trying `data.tilexxxx` first would make it works with it too.

**Comment by YosuaHamonangan** [comment]
oh, I see, make sense, will fix the request

**Comment by obiot** [comment]
👍

**Pull request merged**